yt/visualization/_commons.py

```diff
@@ -93,3 +97,128 @@ def get_canvas(figure, filename):
             f"without an extension."
         )
     return get_canvas_class(suffix)(figure)
+
```

Comment:
Are these functions inspired by or taken from somewhere else?

Reply:
It's all moved code:
- the `get_x_minorticks` routines are taken (moved) from `yt.visualization.plot_container`
-`get_symlog_majorticks` is extracted from `ImagePlotMPL._init_image`
- `get_default_from_config` is adapted from a closure in `PlotContainer.setup_defaults`

The may reason for moving them around was to help me remembering were everything I needed was defined, but I may restore some of them to their original location when I try to clean this branch a bit.